pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -247,7 +247,7 @@ def __load_weights_on_main_process(self) -> None:
         self._model = model
 
     def _close_logger(self, trainer) -> None:
-        if hasattr(trainer, "logger"):
+        if hasattr(trainer, "logger.finalize"):
```

Comment:
how about this?
```suggestion
        if trainer.logger is not None:
```

Reply:
If I set `logger=False`, won't AttributeError be raised as before?

Reply:
If you set logger=False then trainer.logger is set to None.

Reply:
Sorry, thanks